src/libs/print_settings.c

```diff
@@ -0,0 +1,1476 @@
+
+  for(int i=0; i<9; i++)
+  {
+    /* block signal handler */
+    g_signal_handlers_block_by_func (ps->dtba[i],_alignment_callback,user_data);
+
+    if( GTK_WIDGET(ps->dtba[i]) == tb )
+    {
+      gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(ps->dtba[i]),TRUE);
+      index=i;
+    }
+    else gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(ps->dtba[i]),FALSE);
+
+    /* unblock signal handler */
+    g_signal_handlers_unblock_by_func (ps->dtba[i],_alignment_callback,user_data);
+  }
+  ps->prt.page.alignment = index;
+  _update_slider (ps);
+}
+
+static void
+_orientation_changed (GtkWidget *combo, dt_lib_module_t *self)
+{
+  dt_lib_print_settings_t *ps = (dt_lib_print_settings_t *)self->data;
+
+  ps->prt.page.landscape = dt_bauhaus_combobox_get (combo);
+
+  _update_slider (ps);
+}
+
+static void
+_unit_changed (GtkWidget *combo, dt_lib_module_t *self)
+{
+  dt_lib_print_settings_t *ps = (dt_lib_print_settings_t *)self->data;
+
+  ps->unit = dt_bauhaus_combobox_get(combo);
+  dt_conf_set_int("plugins/print/print/unit", ps->unit);
+
+  double margin_top = ps->prt.page.margin_top;
+  double margin_left = ps->prt.page.margin_left;
+  double margin_right = ps->prt.page.margin_right;
+  double margin_bottom = ps->prt.page.margin_bottom;
+
+  const int n_digits = (int)(1.0 / (units[ps->unit] * 10.0));
+
+  gtk_spin_button_set_digits(GTK_SPIN_BUTTON(ps->b_top),    n_digits);
+  gtk_spin_button_set_digits(GTK_SPIN_BUTTON(ps->b_bottom), n_digits);
+  gtk_spin_button_set_digits(GTK_SPIN_BUTTON(ps->b_left),   n_digits);
+  gtk_spin_button_set_digits(GTK_SPIN_BUTTON(ps->b_right),  n_digits);
+
+  const float incr = units[ps->unit];
+
+  gtk_spin_button_set_increments(GTK_SPIN_BUTTON(ps->b_top), incr, incr);
+  gtk_spin_button_set_increments(GTK_SPIN_BUTTON(ps->b_bottom), incr, incr);
+  gtk_spin_button_set_increments(GTK_SPIN_BUTTON(ps->b_left), incr, incr);
+  gtk_spin_button_set_increments(GTK_SPIN_BUTTON(ps->b_right), incr, incr);
+
+  _update_slider (ps);
+
+  // convert margins to new unit
+  gtk_spin_button_set_value(GTK_SPIN_BUTTON(ps->b_top),    margin_top * units[ps->unit]);
+  gtk_spin_button_set_value(GTK_SPIN_BUTTON(ps->b_bottom), margin_bottom * units[ps->unit]);
+  gtk_spin_button_set_value(GTK_SPIN_BUTTON(ps->b_left),   margin_left * units[ps->unit]);
+  gtk_spin_button_set_value(GTK_SPIN_BUTTON(ps->b_right),  margin_right * units[ps->unit]);
+
+}
+
+static void
+_style_callback(GtkWidget *widget, dt_lib_module_t *self)
+{
+  dt_lib_print_settings_t *ps = (dt_lib_print_settings_t *)self->data;
+
+  if(dt_bauhaus_combobox_get(ps->style) == 0)
+  {
+    dt_conf_set_string("plugins/print/print/style", "");
+  }
+  else
+  {
+    const gchar *style = dt_bauhaus_combobox_get_text(ps->style);
+    dt_conf_set_string("plugins/print/print/style", style);
+  }
+
+  if (ps->v_style) g_free(ps->v_style);
+  ps->v_style = g_strdup(dt_conf_get_string("plugins/print/print/style"));
```

Comment:
the g_strdup should be removed, dt_conf_get_string() already returns a newly allocated string.